cmd/validate-yaml/main.go

```diff
@@ -0,0 +1,114 @@
+package validate_yaml
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	pkgExtractor "github.com/datreeio/datree/pkg/extractor"
+	"github.com/datreeio/datree/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+type IPrinter interface {
+	PrintFilename(title string)
+	PrintYamlValidationErrors(validationErrors []error)
+	PrintYamlValidationSummary(passedFiles int, allFiles int)
+	PrintMessage(messageText string, messageColor string)
+}
+
+type IReader interface {
+	FilterFiles(paths []string) ([]string, error)
+}
+
+type IExtractor interface {
+	ExtractConfigurationsFromYamlFile(path string) (*[]pkgExtractor.Configuration, string, *pkgExtractor.InvalidFile)
+}
+
+type ValidateYamlCommandContext struct {
+	Printer   IPrinter
+	Reader    IReader
+	Extractor IExtractor
+}
+
+var YamlNotValidError = errors.New("")
+
+func SetSilentMode(cmd *cobra.Command) {
+	cmd.SilenceUsage = true
+	cmd.SilenceErrors = true
+}
+
+func New(ctx *ValidateYamlCommandContext) *cobra.Command {
+	return &cobra.Command{
+		Use:   "validate-yaml <files>",
+		Short: "validates yaml files structure",
+		Long:  "Validates yaml files <files> structure.",
+		Example: utils.Example(`
+		# Validate yaml files using file path
+		datree validate-yaml kube-prod/deployment.yaml
+
+		# Validate yaml files using glob pattern
+		datree validate-yaml kube-*/*.yaml
+
+		# Validate yaml files by sending manifests through stdin
+		cat kube-prod/deployment.yaml | datree validate-yaml -
```

Comment:
where is the implementation of this logic?

Reply:
you are correct there is none